core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -594,12 +593,10 @@ object Chunk extends CollectorK[Chunk] {
   /** Creates a chunk from a `scala.collection.Iterable`. */
   def iterable[O](i: collection.Iterable[O]): Chunk[O] =
     i match {
-      case ArrayBackedSeq(arr) =>
-        // arr is either a primitive array or a boxed array
-        // cast is safe b/c the array constructor will check for primitive vs boxed arrays
-        array(arr.asInstanceOf[Array[O]])
-      case v: Vector[O]                    => vector(v)
-      case b: collection.mutable.Buffer[O] => buffer(b)
+      case a: collection.mutable.ArraySeq[O]   => arraySeq(a)
+      case a: collection.immutable.ArraySeq[O] => arraySeq(a)
```

Comment:
I have replaced the previous implementation of this behaviour via `ArrayBackedSeq` (now deleted).